source/develop/release-management/features/external-network-provider.html.md

```diff
@@ -0,0 +1,456 @@
+category: feature
+authors: mmirecki
+wiki_category: Feature
+wiki_title: Features/External Network Providers
+wiki_revision_count: 1
+wiki_last_updated: 2016-01-25
+feature_name: External Network Providers
+feature_modules: engine,vdsm
+feature_status: In Development
+---
+
+# External Network Providers
+
+
+## Summary
+
+Currently it is possible to integrate oVirt with Openstack Neutron (using  the
+"Openstack Neutron Integration" feature). The purpose of this feature is to
+simplify this and to make it easier to add support for networks supplied by
+external providers. The providers will be able to provide networks using a
+simplified Neutron-like API.
+
+
+### Owner
+
+*   Feature Owner: Marcin Mirecki: [ mmirecki](User:mmirecki)
+
+<!-- -->
+
+*   Email: <mmirecki@redhat.com>
+
+### Benefit to oVirt
+
+oVirt VMs will be able to use networks definded by external providers.
+oVirt will provide an API for the external providers (vendors Software Defined
+Networking products) allowing them to easily integrate with oVirt.
+
+
+## Detailed Description
+
+Many organizations use centralized network management systems to handle all
+their networking, and would like them to manage the network topology of
+their oVirt environments. 
+Currently the only option of using external networks in oVirt is to used the
+OpenStack Neutron integration, which allows the import of OpenStack Neutron
+networks and the provisioning of VM's connected to these networks. 
+This feature is however very specific to OpenStack Neutron. 
+We would like to extend this functionality to other external network providers,
+by making it simpler, more general and less dependant on OpenStack Neutron features. 
+The result should be an API over which oVirt can communicate with external
+network management systems, and use the networks defined in them in provisioned VMs.
+
+For information about OpenStack Neutron integration please refer to:
+[OSN Integration](http://www.ovirt.org/develop/release-management/features/network/osn-integration/)
+[Detailed OSN Integration](http://www.ovirt.org/develop/release-management/features/network/detailed-osn-integration/)
+
+### Differences to OpenStack Neutron Integration
+
+*   hiding of UI features related to OpenStack Neutron, such as tenants and OpenStack Neutron driver details
+![](external_network_provider_import_dialog_changes.png "fig:external_network_provider_import_dialog_changes.png")
+
+*   simplified REST API. Base on the OpenStack Neutron API, but simplified to contain
+only a subset of its schema which is relevant to the external network providers.
+This will require a rework of the REST API layer.
+
+*   read-only mode for external network
+
+*   (optionally) read-only mode for subnets
+
+*   VIF driver - the implementation of the VIF driver is now completely the responsibility 
+of the external network provider
+
+*   No OpenStack Neutron agent - since the VIF driver implementation is provider dependent,
```

Comment:
No *special handling of* OpenStack Neutron agent.

(we don't mind it if a partner uses an agent or not)